source/postprocess/geoid.cc

```diff
@@ -489,98 +511,104 @@ namespace aspect
+              for (unsigned int ideg =  min_degree; ideg < max_degree+1; ideg++)
+                {
+                  for (unsigned int iord = 0; iord < ideg+1; iord++)
+                    {
+                      output_surface_dynamic_topo_contribution_SH_coes << ideg
+                                                                       << ' '
+                                                                       << iord
+                                                                       << ' '
+                                                                       << surface_dyna_topo_contribution_coecos.at(SH_coes_ind)
+                                                                       << ' '
+                                                                       << surface_dyna_topo_contribution_coesin.at(SH_coes_ind)
+                                                                       << std::endl;
+                      ++SH_coes_ind;
+                    }
+                }
+
+              const std::string surface_dynamic_topo_contribution_SH_coes_filename = this->get_output_directory() +
+                                                                                     "surface_dynamic_topography_contribution_SH_coefficients." +
+                                                                                     dealii::Utilities::int_to_string(this->get_timestep_number(), 5);
+
+              // Because each processor already held all the SH coefficients from surface dynamic topography contribution, we only need to stop by the processor 0
+              // to get the data. On processor 0, collect all the data and put them into the output surface dynamic topography contribution SH coefficients file.
+              if (dealii::Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0)
+                {
+                  std::ofstream surface_dynamic_topo_contribution_SH_coes_file (surface_dynamic_topo_contribution_SH_coes_filename.c_str());
+                  surface_dynamic_topo_contribution_SH_coes_file << "# "
+                                                                 << "degree order cosine_coefficient sine_coefficient"
+                                                                 << std::endl;
+                  std::ostringstream output_surface_delta_rho;
+                  output_surface_delta_rho << surface_delta_rho;
+                  surface_dynamic_topo_contribution_SH_coes_file << "surface density contrast(kg/m^3): "
+                                                                 << output_surface_delta_rho.str()
+                                                                 << std::endl;
+                  // write out the data on processor 0
+                  surface_dynamic_topo_contribution_SH_coes_file << output_surface_dynamic_topo_contribution_SH_coes.str();
+                }
+            }
+        }
+
+      if (remove_dynamic_topo_contribution == false)
```

Comment:
I think you can remove this `if` and the braces in line 562 above and 565 below. The result should be the same.